### DISCUSSION TITLE
Disable layout for routes

### DISCUSSION THREAD

**Discussion by folamy** [hannoeru/vite-plugin-pages#355] [Q&A]
### Description

I would like to disable layout for some certain routes, like the landing page.
Is that possible?

### Suggested solution

```js
<route lang="yaml">
name: Home
meta:
  layout: false
</route>
```

### Validations

- [X] Follow the [Code of Conduct](https://github.com/hannoeru/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the docs.
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by abdul-alhasany** [comment]
Are you using this plugin: [vite-plugin-vue-layouts](https://github.com/JohnCampionJr/vite-plugin-vue-layouts)?

Otherwise vite-plugin-pages does does not have a layout option. You can, however, use the meta value to manipulate your component. 

For example:
In your component you add the code you had:
```vue
<route lang="yaml">
name: Home
meta:
  layout: false
</route>
```

And then when route loads the parent component (where <router-view /> is ) you can do something like this:
```js
const shouldShowLayout = computed(() => {
	const { meta } = useRoute();
	return meta.layout !== false;
});
```

I might be able to help more if more code was provided.
